src/ert/_c_wrappers/job_queue/workflow_job.py

```diff
@@ -137,30 +136,20 @@ def run(self, ert: "EnKFMain", arguments: List[Any], verbose: bool = False) -> A
         if self.internal and self.script is not None:
             script_obj = ErtScript.loadScriptFromFile(self.script)
             self.__script = script_obj(ert)
-            result = self.__script.initializeAndRun(
-                self.argumentTypes(), arguments, verbose=verbose
-            )
-
-        elif self.internal and self.script is None:
```

Comment:
The if dropped `self.script is None` here. What is the implication of that.

Reply:
Very good question, I think the implication of dropping the `self.script is None` is that nothing changes.

Let's try to reason through this beautifully designed if, elfi chain of logic:

For the sake of being less verbose, I will assume `self.internal` and `self.script` are replaced by internal and script:

We have 
```
If internal and script is not None:
   case1()
elif internal and script is None:
   case2()
elif not internal:
   case3()
else:
   case4()
```


|   internal |   script	|   case	|  
|---	|---	|---	|
|   True	|   True	|   1	|
|   True	|   False	|   2	|
|   False	|   True	|   3	|
|   False	|   False	|   3	||


So what do we learn from the table above? 

1. That if `internal` is `False` we are always running `case3`  and it does not really matter if `script` is `True` or `False`.
2. We never reach `case4` hence we never raise that very useful `UserWarning("Unknown script type")` [Something I just realized by making the table above and maybe I should remove that final `else` if it is unreachable code]
3. It is redundant to check if the `script` is `False` to verify if we need to run `case2` because  if `internal` is `True` and we got to check if we need to run `case2` that means `script` definitely is `False` so we don't need to check it